test/e2e/pkg/utils.go

```diff
@@ -0,0 +1,282 @@
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"strconv"
+
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// proTxHashGenerator generates pseudorandom proTxHash based on a seed.
+type proTxHashGenerator struct {
+	random *rand.Rand
+}
+
+func newProTxHashGenerator(seed int64) *proTxHashGenerator {
+	return &proTxHashGenerator{
+		random: rand.New(rand.NewSource(seed)), //nolint: gosec
+	}
+}
+
+func (g *proTxHashGenerator) generate() crypto.ProTxHash {
+	seed := make([]byte, crypto.DefaultHashSize)
+	_, err := io.ReadFull(g.random, seed)
+	if err != nil {
+		panic(err) // this shouldn't happen
+	}
+	return seed
+}
+
+// quorumHashGenerator generates pseudorandom quorumHash based on a seed.
+type quorumHashGenerator struct {
+	random *rand.Rand
+}
+
+func newQuorumHashGenerator(seed int64) *quorumHashGenerator {
+	return &quorumHashGenerator{
+		random: rand.New(rand.NewSource(seed)),
```

Comment:
Please address linter alert below

Reply:
it is easier to disable linting for this line